gemini/data-generation/synthetic_data_generation/plugins/Gemini.py

```diff
@@ -0,0 +1,80 @@
+"""
+Implements the plugin for interacting with Gemini.
+"""
+
+import logging
+
+import jinja2
+from snowfakery.plugins import SnowfakeryPlugin
+from vertexai.generative_models import GenerativeModel
+
+
+class Gemini(SnowfakeryPlugin):
+    """
+    Plugin for interacting with Gemini.
+    """
+
+    class Functions:
+        """
+        Functions to implement field / object level data generation
+        """
+
+        def fill_prompt(self, prompt_name, **kwargs):
+            """
+            Returns a formatted prompt
+            """
+            return (
+                jinja2.Environment(
+                    loader=jinja2.FileSystemLoader(
+                        searchpath="./synthetic_data_generation/prompts"
+                    )
+                )
+                .get_template(prompt_name)
+                .render(**kwargs)
+            )
+
+        def generate(self, prompt_name, temperature=0.9, top_p=1, **kwargs):
```

Comment:
Maybe make the model name a parameter to allow swapping out with other gemini versions

Reply:
Implemented in the new notebook.